### PR TITLE
SDK-5621 Remove integrator lock

### PR DESCRIPTION
**Pull request by pavelmaksimov25**
Ticket: https://spryker.atlassian.net/browse/SDK-5621

### CODE REVIEW

File: src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsExecutionDto
+     *
+     * @return \Upgrade\Application\Dto\StepsResponseDto
+     */
+    public function run(StepsResponseDto $stepsExecutionDto): StepsResponseDto
+    {
+        if ($this->filesystem->exists(static::FILENAME)) {
```
Comment (sergeyspryker) [inline src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php:49]: Why do we need this check? Just remove it

File: src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            $this->filesystem->remove(static::FILENAME);
+        }
+
+        if ($this->vsc->hasUncommittedFile(static::FILENAME)) {
+            $this->vsc->removeTrackedFiles(static::FILENAME);
+            $this->vsc->commitWithMessage('Remove integrator.lock file');
```
Comment (sergeyspryker) [inline src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php:55]: All the files before added in previous steps are gonna be commited with this message

**Review by sergeyspryker — CHANGES_REQUESTED** [review]

File: src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php (hunk 1 above)
Comment (pavelmaksimov25) [reply src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php:49]: to remove a file only if it exists. Why?

File: src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php (hunk 2 above)
Comment (pavelmaksimov25) [reply src/Upgrade/Application/Strategy/Common/Step/RemoveIntegratorLockFileStep.php:55]: During testing, I saw only integrator lock changes. 
However, all the changes are committed before this step and here we commit only specific file, so shouldn't cause any problems.

File: src/Upgrade/Infrastructure/VersionControlSystem/Git/Adapter/GitAdapter.php [hunk 3]
```diff
@@ -58,6 +58,16 @@ public function hasAnyUncommittedChanges(StepsResponseDto $stepsExecutionDto): S
         return $this->git->hasAnyUncommittedChanges($stepsExecutionDto);
     }
 
+    /**
+     * @param string $filename
+     *
+     * @return bool
+     */
+    public function hasUncommittedFile(string $filename): bool
```
[the remark this answers is not in the document]
Comment (pavelmaksimov25) [reply src/Upgrade/Infrastructure/VersionControlSystem/Git/Adapter/GitAdapter.php:66]: Agree. The GitAdapter is kind of an obsolete thing here

**Pull request merged**